src/interfaces/ResourceInterface.ts

```diff
@@ -0,0 +1,26 @@
+import StoreData from './StoreData'
+
+/**
+ * Generic interface for a standalone ResourceInterface (e.g. a HAl resource with an own store entry and a self link)
+ * Can be a collection or a single entity
+ */
+interface ResourceInterface {
+    _meta: {
+        self: string | null
+        load: Promise<ResourceInterface>
+        loading: boolean
+        deleting?: boolean
+    }
+
+    _storeData?: StoreData
+
+    $reload: () => Promise<ResourceInterface>
+    $post: (data: unknown) => Promise<ResourceInterface | null>
+    $patch: (data: unknown) => Promise<ResourceInterface>
+    $del: () => Promise<string | void>
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    $href: (relation: string, templateParams: any) => Promise<string | undefined>
```

Comment:
According to https://github.com/bramstein/url-template/blob/main/lib/url-template.d.ts#L2, the type of `templateParams` could be specified to `Record<string, string | number | boolean>` (an update of the url-template library might be necessary, idk)

Reply:
Upgrade to url-template v3 didn't work out of the box. I've included the type definition but kept the version at 2.0.8 for now.